src/r5py/r5/transport_network.py

```diff
@@ -169,10 +144,15 @@ def _working_copy(self, input_file):
         destination_file = pathlib.Path(
             self._cache_directory / input_file.name
         ).absolute()
-        try:
-            destination_file.symlink_to(input_file)
-        except OSError:
-            shutil.copyfile(str(input_file), str(destination_file))
+
+        with filelock.FileLock(
+            destination_file.parent / f"{destination_file.name}.lock"
+        ):
+            if not destination_file.exists():
+                try:
+                    destination_file.symlink_to(input_file)
+                except OSError:
+                    shutil.copyfile(str(input_file), str(destination_file))
```

Comment:
One and the same `TransportNetwork` might be used in several threads at the same time. This `filelock.FileLock` makes sure we don’t fail copying the temp file due to race conditions